src/fileops.rs

```diff
@@ -31,6 +32,13 @@ impl Fd {
       ret @ _ => Ok(ret)
     }
   }
+
+  pub fn write<T>(&self, t: T) -> Result<size_t, Error> where T: Sized {
+    match unsafe { libc::write(self.0, &t as *const T as *const c_void, size_of::<T>() as size_t) } {
```

Comment:
yeah..... idk whats going to happen. im supposed to hand over a void pointer to muh struct